mysql/datadog_checks/mysql/mysql.py

```diff
@@ -280,9 +280,11 @@ class MySql(AgentCheck):
     SLAVE_SERVICE_CHECK_NAME = 'mysql.replication.slave_running'
     DEFAULT_MAX_CUSTOM_QUERIES = 20
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
-        self.mysql_version = {}
+    def __init__(self, name, init_config, instances=None):
+        AgentCheck.__init__(self, name, init_config, instances)
+        self.version = ''
```

Comment:
Probably worth creating a named tuple to store the version rather than having three different attributes. Something like:
```
MySQLVersion = namedtuple('MySQLVersion', ['semver', 'flavor', 'build'])
````
And then in the constructor:
```
self._version = None
```

Then you can replace the _get_version method for:
```
@property
def version():
    if not self._version:
        <current _ get_version code>
        <set_metadata>
        self._version = MySQLVersion(parts[0], build, flavor)
    return self._version
```

That way you don't to include all the version parsing logic into _collect_metadata so you have more atomic methods.
You can find a similar pattern in https://github.com/DataDog/integrations-core/pull/4844/files#diff-5557a2bc9bebe6d84b8a6fa72246b83bR133